### ISSUE TITLE
Export categories from parent classes are not reloading, and cause "duplicate categories".

### ISSUE THREAD

**Issue by Frontrider** [godotengine/godot#71616]
### Godot version

4.0 beta 13

### System information

Manjaro Linux

### Issue description

Export categories from the parent script are not shown unless you reload the project, and can cause the same category name to be duplicated.

The first one is a bug, that needs to be fixed. Categories shouldn't just vanish like that.

The second one is a thing I can't place, personally I'd expect those to be merged instead of getting 2 menus with the same name.

### Steps to reproduce

If you extend this script:
```
extends CharacterBody2D
class_name Speed

@export_category("speed")
@export var speed:int = 1.0:
	set(new_value):
		speed = new_value
	get:
		if(parent_segment != null):
			return parent_segment.speed
		return speed
```
With this:
```
extends Speed

@export_category("speed")
@export var turn_speed = 10

```
Then at first you won't see the category from the parent class, then after a project reload you'll see 2 categories with the name "speed".

### Minimal reproduction project

N/A

**Comment by KoBeWi** [comment]
It might be related to other script reloading issues. Strange that reloading scene does not help; it usually works.

> The second one is a thing I can't place, personally I'd expect those to be merged instead of getting 2 menus with the same name.

The properties are grouped by class first and then by category. That's how inspector is designed; it does not support merging categories between classes.

**Comment by vnen** [comment]
@YuriSizov is there anything we can do about this? Potentially we could check for a probable misuse of the feature, but all in all it seems to be working as intended.

**Comment by Frontrider** [comment]
Can you show an error if the group was duplicated? I think that should be more than enough to "enforce the correct usage", and not have anyone mention it again. :)

**Comment by YuriSizov** [comment]
We could add a warning if you try to add a category or a (sub)group with the same signature, but otherwise indeed this is working as intended and grouping properties can be duplicated without any repercussions. You just shouldn't expect them to merge.

I guess a warning would be like "child category is shadowing parent category, pay attention"? May want to wait a bit and see if there is a lot of confusion, or of it's just this one report.

**Comment by Frontrider** [comment]
> I guess a warning would be like "child category is shadowing parent category, pay attention"? May want to wait a bit and see if there is a lot of confusion, or of it's just this one report.

That would be perfect. After the explanation the only issue I see here is that it *looks like* a bug.

**Comment by dalexeev** [comment]
This needs to be re-tested, maybe something has changed after #78254 and #81221.

**Comment by KoBeWi** [comment]
> Then at first you won't see the category from the parent class, then after a project reload you'll see 2 categories with the name "speed".

The double category shows immediately now
![image](https://github.com/godotengine/godot/assets/2223172/3eeda7bb-66d1-4f53-a01d-67c2a6c9b8c7)

**Comment by dalexeev** [comment]
> Export categories from the parent script are not shown unless you reload the project, and can cause the same category name to be duplicated.
> 
> The first one is a bug, that needs to be fixed. Categories shouldn't just vanish like that.

> It might be related to other script reloading issues. Strange that reloading scene does not help; it usually works.

> The double category shows immediately now

I think the first part regarding script reloading has been fixed, probably by #83123.

> The second one is a thing I can't place, personally I'd expect those to be merged instead of getting 2 menus with the same name.

> The properties are grouped by class first and then by category. That's how inspector is designed; it does not support merging categories between classes.

I think after a number of PRs this now works properly and categories/groups/subgroups don't conflict with each other and regular properties:

* #78254
* #81221
* #87915
* #88318
* #89631

![](https://github.com/godotengine/godot/assets/47700418/46d28c0f-2fa0-4e80-b7ca-e92300b67843)

> is there anything we can do about this? Potentially we could check for a probable misuse of the feature, but all in all it seems to be working as intended.

> Can you show an error if the group was duplicated?

> We could add a warning if you try to add a category or a (sub)group with the same signature, but otherwise indeed this is working as intended and grouping properties can be duplicated without any repercussions.

> I guess a warning would be like "child category is shadowing parent category, pay attention"?

I think this is no longer necessary since you can now have categories/groups/subgroups duplicated at different inheritance levels.

So, if there is no objection, I suggest closing this issue after merging #89631.

**Comment by Frontrider** [comment]
IF nobody has anything else to add, close it.
